src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java

```diff
@@ -64,7 +64,16 @@ FilterVariable generateFilter(SnapshotBuilderProgramDataRangeCriteria rangeCrite
                 new Literal(rangeCriteria.getHigh()))));
   }
 
+   FilterVariable selectAllValuesOfListCriteria() {
+    return platform -> "1=1";
```

Comment:
This could also be `BooleanAndOrFilterVariable(BooleanAndOrFilterVariable.LogicalOperator.AND, List.of()`

Reply:
Another way of writing this is:

```
FilterVariable selectAllValuesOfListCriteria() {
    return new FilterVariable() {
      @Override
      public String renderSQL(CloudPlatformWrapper platform) {
        return "1=1";
      }
    };
  }
```

I wanted to use a `lambda` to clean this up, I made this into a function to make whats going on more readable. 

Hopefully others will see this function, and read `selectAllValuesOfListCriteria `? Does this make sense?

Reply:
This form looks fine to me. Java infers that an interface is a `@FunctionalInterface` if it has a single abstract method so using a lambda for a class in this context is ok.

Reply:
Another way to write this is to declare a method that matches `FilterVariable` and then refer to the method, instead of calling a function that returns a new method each time.
```
  String selectAllValuesOfListCriteria(CloudPlatformWrapper platform) {
      return "1=1";
  }

[...]

    if (listCriteria.getValues() == null || listCriteria.getValues().isEmpty()) {
      return this::selectAllValuesOfListCriteria;
    }
```